freqtrade/optimize/backtesting.py

```diff
@@ -457,13 +455,21 @@ def backtest_one_strategy(self, strat: IStrategy, data: Dict[str, Any], timerang
         preprocessed = self.strategy.ohlcvdata_to_dataframe(data)
 
         # Trim startup period from analyzed dataframe
-        for pair, df in preprocessed.items():
-            preprocessed[pair] = trim_dataframe(df, timerange,
-                                                startup_candles=self.required_startup)
-        min_date, max_date = history.get_timerange(preprocessed)
-        if min_date is NaT or max_date is NaT:
+        for pair in list(preprocessed.keys())[:]:
+            df = preprocessed[pair]
+            df = trim_dataframe(df, timerange, startup_candles=self.required_startup)
+            if len(df) > 0:
+                preprocessed[pair] = df
+            else:
+                logger.warning(f'{pair} has no data left after adjusting for startup candles, '
+                               f'skipping.')
+                del preprocessed[pair]
+
+        if len(preprocessed) == 0:
```

Comment:
```suggestion
        if preprocessed:
```
Since `preprocessed` looks like a dictionary, this is all it takes to check if it's empty.